bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/discovery/SurePetcareDiscoveryService.java

```diff
@@ -0,0 +1,176 @@
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        ScheduledFuture<?> job = discoveryJob;
+        if (job != null) {
+            job.cancel(true);
+            discoveryJob = null;
+            logger.debug("Stopped Sure Petcare device background discovery");
+        }
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting Sure Petcare discovery scan");
+        // If the bridge is not online no other thing devices can be found, so no reason to scan at this moment.
+        removeOlderResults(getTimestampOfLastScan());
+        if (bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
+            logger.debug("Starting device discovery for bridge {}", bridgeUID);
+            bridgeHandler.listHouseholds().forEach(this::householdDiscovered);
+            bridgeHandler.listPets().forEach(this::petDiscovered);
+            bridgeHandler.listDevices().forEach(this::deviceDiscovered);
+        }
+    }
+
+    private void householdDiscovered(SurePetcareHousehold household) {
+        logger.debug("Discovered household: {}", household.name);
+        ThingUID thingsUID = new ThingUID(THING_TYPE_HOUSEHOLD, bridgeUID, household.id.toString());
+        Map<String, Object> properties = new HashMap<String, Object>(household.getThingProperties());
+        thingDiscovered(DiscoveryResultBuilder.create(thingsUID).withLabel(household.name).withProperties(properties)
+                .withRepresentationProperty(PROPERTY_NAME_ID).withBridge(bridgeUID).build());
+    }
+
+    private void petDiscovered(SurePetcarePet pet) {
+        logger.debug("Discovered pet: {}", pet.name);
+        ThingUID thingsUID = new ThingUID(THING_TYPE_PET, bridgeUID, pet.id.toString());
+        Map<String, Object> properties = new HashMap<String, Object>(pet.getThingProperties());
+        thingDiscovered(DiscoveryResultBuilder.create(thingsUID).withLabel(pet.name).withProperties(properties)
+                .withRepresentationProperty(PROPERTY_NAME_ID).withBridge(bridgeUID).build());
+    }
+
+    private void deviceDiscovered(SurePetcareDevice device) {
+        logger.debug("Discovered device: {}", device.name);
+        ThingTypeUID typeUID = null;
+        switch (ProductType.findByTypeId(device.productId)) {
+            case HUB:
+                typeUID = THING_TYPE_HUB_DEVICE;
+                break;
+            case CAT_FLAP:
+                typeUID = THING_TYPE_FLAP_DEVICE;
+                break;
+            case PET_FLAP:
+                typeUID = THING_TYPE_FLAP_DEVICE;
+                break;
+            case PET_FEEDER:
+                typeUID = THING_TYPE_FEEDER_DEVICE;
+                break;
+            case UNKNOWN:
+            default:
+                return;
+        }
+        ThingUID thingsUID = new ThingUID(typeUID, bridgeUID, device.id.toString());
+        Map<String, Object> properties = new HashMap<String, Object>(device.getThingProperties());
```

Comment:
Please use the diamond operator, ie replace the second occurrence of the same generic by `<>`

Reply:
done